src/screens/social/NotificationsScreen.tsx

```diff
@@ -1,51 +1,110 @@
 import { SocialRoute } from '@nightlight/src/types';
 import React, { useEffect, useState } from 'react';
-import { SafeAreaView, ScrollView, Text, View } from 'react-native';
+import { SafeAreaView, ScrollView, Text, View, FlatList } from 'react-native';
 import NotificationsScreenStyles from './NotificationsScreen.styles';
-import { TEST_NOTIFICATIONS } from '@nightlight/src/testData';
 import NotificationCard from '@nightlight/components/social/NotificationCard';
+import { useAuthContext } from '@nightlight/src/contexts/AuthContext';
+import { customFetch } from '@nightlight/src/api';
+import NotificationCardStyles from '@nightlight/components/social/NotificationCard.styles';
 const NotificationsScreen = () => {
-  const [notifications, setNotifications] = useState(TEST_NOTIFICATIONS);
+  const [notifications, setNotifications] = useState([]);
   const [counter, setCounter] = useState(0);
-  //TODO: need to pull notifications from backend (getNotifications)
 
-  //   useEffect(() => {
-  //     setCounter(notifications.length)
-  //   });
+  // user id
+  const { userDocument } = useAuthContext();
+  let userid = userDocument?._id;
 
-  // called when there are no active group
+  useEffect(() => {
+    customFetch({
+      resourceUrl: `/notifications/?userId=${userid}`,
+      options: {
+        method: 'GET',
+      },
+    })
+      .then(res => {
+        let count = 0;
+        res.notifications.forEach(
+          (item: { data: { notificationType: string } }) => {
+            if (
+              item.data.notificationType === 'friendRequest' ||
+              item.data.notificationType === 'groupInvite'
+            ) {
+              count++;
+            }
+          }
+        );
+        setCounter(count);
```

Comment:
Might be more efficient to use `.filter()` then `.length` :) You might also wanna consider defining an array of notification types to count `[NotificationType.FriendRequest, NotificationType.GroupInvite]`, and simply check membership of each notification type using `.includes()`.